localstack/services/plugins.py

```diff
@@ -542,24 +547,32 @@ def get_state(self, name: str) -> Optional[ServiceState]:
         return ServiceState.AVAILABLE
 
     def get_service_container(self, name: str) -> Optional[ServiceContainer]:
-        container = super().get_service_container(name)
-        if container:
+        if container := self._services.get(name):
             return container
 
         if not self.exists(name):
             return None
 
-        # this is where we start lazy loading. we now know the PluginSpec for the API exists,
-        # but the ServiceContainer has not been created
-        plugin = self._load_service_plugin(name)
-        if not plugin or not plugin.service:
-            return None
+        load_lock = self._plugin_load_locks[name]
+        with load_lock:
+            # check once again to avoid race conditions
+            if container := self._services.get(name):
+                return container
 
-        with self._mutex:
-            if plugin.service.name() not in self._services:
-                super().add_service(plugin.service)
+            # this is where we start lazy loading. we now know the PluginSpec for the API exists,
+            # but the ServiceContainer has not been created.
+            # this control path will be executed once per service
+            plugin = self._load_service_plugin(name)
+            if not plugin or not plugin.service:
+                return None
+
+            with self._mutex:
+                if plugin.service.name() not in self._services:
```

Comment:
Wondering if there can be a situation where this condition is _not_ true, i.e., where an entry for this service name would already exist in `self._services` (since we're now locking on the service `name` in this critical section, and `_load_service_plugin(..)` in line 565 above is side-effect free). 

The only case where this could happen is if `plugin.service.name()` returns a string that is different from `name` - can this be the case?

If that's something we're not expecting to happen, maybe we could proactively raise an error, to avoid such cases from going unnoticed:
```
            with self._mutex:
                if plugin.service.name() in self._services:
                    raise ServiceStateException(
                        "Service {plugin.service.name()} unexpectedly already in list of loaded services")
                    )
                super().add_service(plugin.service)
```

Reply:
good point, can't happen anymore. fixed!